src/pyodide/internal/tarfs.js

```diff
@@ -0,0 +1,86 @@
+import { default as TarReader } from "pyodide-internal:packages_tar_reader";
+
+export function createTarFS(Module) {
+  const FS = Module.FS;
+  const TARFS = {
+    mount(mount) {
+      return TARFS.createNode(null, "/", mount.opts.info);
+    },
+    createNode(parent, name, info) {
+      let mode = info.mode;
+      const isDir = !!info.children;
```

Comment:
Nitpicking: could you be more specific with your check here to make it clearer? i.e. `info.children !== undefined`.